### PR TITLE
Adds a couple of read filters to the ExomeReadCounts tool.

### PR DESCRIPTION
**Pull request by vruano**
Adds a couple of read filters to the ExomeReadCounts tool.  
  
  - Don't count reads marked as duplicates
  - Don't count reads that do not have any single base aligned versus the reference.

This addresses issues #484 and #485.

### CODE REVIEW

**Comment by vruano** [comment]
@akiezun please review.

File: src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java [hunk 1]
```diff
@@ -21,6 +22,7 @@ private ReadFilterLibrary(){ /*no instance*/ }
     public static final ReadFilter MAPPING_QUALITY_NOT_ZERO = read -> read.getMappingQuality() != 0;
     public static final ReadFilter READLENGTH_EQUALS_CIGARLENGTH = read -> read.getReadLength() == read.getCigar().getReadLength();
     public static final ReadFilter GOOD_CIGAR =  read -> CigarUtils.isGood(read.getCigar());
+    public static final ReadFilter NON_ZERO_REFERENCE_LENGTH_ALIGNMENT = read -> read.getCigar().getCigarElements().stream().anyMatch(c -> c.getOperator().consumesReferenceBases() && c.getLength() > 0);
```
Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java]: this new filter needs tests in ReadFilterLibraryUnitTest

**Comment by akiezun** [comment]
back to @vruano - please add tests for the new filter

**Comment by vruano** [comment]
Done, back to @akiezun.

**Comment by akiezun** [comment]
tests look good. not sure why the "travis-ci/push" build fails. good to merge when that is resolved

**Comment by vruano** [comment]
Rerun the test and it worked.

**Pull request merged**